packages/libs/components/src/map/SemanticMarkers.tsx

```diff
@@ -149,12 +155,30 @@ export default function SemanticMarkers({
     }
   }, [animation, map, markers, prevMarkers, recenterMarkers]);
 
+  // remove any selectedMarkers that no longer exist in the current markers
+  useEffect(() => {
+    if (setSelectedMarkers && selectedMarkers) {
+      const prunedSelectedMarkers = selectedMarkers.filter((id) =>
+        consolidatedMarkers.find(({ props }) => id === props.id)
+      );
+
+      if (prunedSelectedMarkers.length < selectedMarkers.length)
+        setSelectedMarkers(prunedSelectedMarkers);
+    }
+  }, [consolidatedMarkers, selectedMarkers]);
```

Comment:
I think using `useEffect` here is fine. However, I'm not convinced that we should do this (remove selected markers when they go offscreen). This means that a user cannot select markers that fall outside of the users viewport size, _at a given geohash_. Would it be "good enough" to clear marker selection when the geohash changes (and this would happen in the MapAnalysis component)?

Reply:
If we keep them when they go off-screen then it's even more important to put something in the header to say how many things are currently selected. I'm not saying that's a bad idea, but it needs further UI thought: can you cancel your selections from the header also?

The use case for maintaining selections outside the viewport is presumably so the user can zoom into Australia, pick a few markers, pan over to South America and pick a few more.

At the moment the plan is to _add_ the marker-selection as a filter to the floating viz filters. The floating viz filters already include the viewport, though of course that could be removed if a selected-marker-based filter is being applied. 

I personally think the UX is a lot simpler if we restrict everything to what is currently on-screen/in-viewport.

If the current whole-world/continent leaflet zoom level isn't giving enough geohash resolution, there is always this ticket to help: https://github.com/VEuPathDB/web-monorepo/issues/449 (I have moved it to the new board)

Reply:
I'm half wondering if we should be selecting gehash rectangles instead of markers. Perhaps with shift-click or ctrl-click. It feeds into the discussion of little filters and floating visualizations here: https://github.com/VEuPathDB/web-monorepo/issues/596

Reply:
> I'm half wondering if we should be selecting gehash rectangles instead of markers. Perhaps with shift-click or ctrl-click. It feeds into the discussion of little filters and floating visualizations here: #596 (see point 3.)

I was just thinking the same thing!

Reply:
I'm convinced about removing the selection.